example/builder/src/basic_with_state.dart

```diff
@@ -2,13 +2,15 @@ import 'package:over_react/over_react.dart';
 import 'props_mixin.dart' as pm;
 import 'state_mixin.dart';
 
+// ignore_for_file: uri_has_not_been_generated
 part 'basic_with_state.over_react.g.dart';
 
-@Factory()
-UiFactory<BasicProps> Basic = _$Basic;
+UiFactory<BasicProps> Basic = _$Basic; // ignore: undefined_identifier
 
-@Props()
-class _$BasicProps extends UiProps with pm.ExamplePropsMixinClass {
+class BasicProps = UiProps with pm.ExamplePropsMixin, BasicPropsMixin;
+class BasicState = UiState with ExampleStateMixin, BasicStateMixin;
+
+mixin BasicPropsMixin on UiProps, pm.ExamplePropsMixin {
```

Comment:
I think I like the use of mixins better the more I think about it.  I went with `implements` because I was seeing analysis errors when adding the mixins to the concrete class - but I now see that its just because the mixins were in the wrong order.

Reply:
@greglittlefield-wf now that I've tried this out in WSD though - I think some changes will likely be required in the builder to make it work at build time.  Analysis is clean, but the mixins don't all get copied over correctly:

```dart
mixin FooPropsMixin on UiProps, BarPropsMixin {
  // ...
}
```

__generates:__

```dart
abstract class _$$FooPropsMixin extends UiProps
    with
        FooPropsMixin,
        $FooPropsMixin // If this generated mixin is undefined, it's likely because InputDisplayPropsMixin is not a valid `mixin`-based props mixin, or because it is but the generated mixin was not exported. Check the declaration of InputDisplayPropsMixin.
{
  // ...
}
```

Which will cause a build failure of the form:

```
Error: 'UiProps' doesn't implement 'BarPropsMixin' so it can't be used with 'FooPropsMixin'.
```

Reply:
Addressed via https://github.com/Workiva/over_react/pull/481/commits/a6ed87eba8b1a2b5e8ef16ee753614bb1f8864a9